File: routes/delete_container.go
package routes

import (
	"fmt"
	"os"
	"os/exec"
	"server/method"
	"strings"

	"github.com/gofiber/fiber/v2"
)

func DeleteContainer(app *fiber.App) {

	app.Get("/api/instance/delete/:bot_id/:admin_key", func(c *fiber.Ctx) error {

		var bot_id = c.Params("bot_id")
		var admin_key = c.Params("admin_key")

		if !method.ValidateAdminKey(admin_key) {
			fmt.Println("[Delete] Erreur admin_key", admin_key, " n'est pas valide!")
			return c.Status(fiber.StatusBadRequest).SendString("Invalid admin_key!")
		}

		if _, err := os.Stat(method.PathResolve(method.ProcessCWD(), "ownihrz", bot_id)); os.IsNotExist(err) {
			fmt.Println("[Delete] Erreur bot_id", bot_id, " n'existe pas!")
			return c.Status(fiber.StatusBadRequest).SendString("Invalid bot_id!")
		}

		cliArray := []struct {
			L   string
			CWD string
		}{
			{
				L:   strings.Replace("pm2 stop {Code} -f", "{Code}", bot_id, 1),
				CWD: method.PathResolve(method.ProcessCWD(), "ownihrz", bot_id),
			},

			{
				L:   strings.Replace("pm2 delete {Code}", "{Code}", bot_id, 1),
				CWD: method.PathResolve(method.ProcessCWD(), "ownihrz", bot_id),
			},

			{
				L:   strings.Replace("rm -r --interactive=never {Code}", "{Code}", bot_id, 1),
				CWD: method.PathResolve(method.ProcessCWD(), "ownihrz"),
			},
		}

		for _, index := range cliArray {
			cmd := exec.Command("sh", "-c", index.L)
			cmd.Dir = index.CWD
			cmd.Stdout = os.Stdout
			cmd.Stderr = os.Stderr
			if err := cmd.Run(); err != nil {
				fmt.Print(err)
				return c.Status(fiber.StatusInternalServerError).SendString(err.Error())
			}
		}

		return c.SendStatus(200)
	})
}

File: src/Routes/v1/delete_container.ts
import { validateAdminKey } from '../../method/validateData.js';
import config from '../../method/getConfigData.js';

import { Request, Response } from 'express';
import fs from "node:fs";
import { execSync } from "child_process";
import path from "node:path";
import { db } from '../../method/database.js';
import { getOwnerByCode } from '../../method/getOwnerByCode.js';
import logsRequest from '../../method/logRequest.js';

export const route = {
    type: 'get',
    apiPath: '/api/v1/instance/delete/:bot_id/:admin_key',
    run: async (req: Request, res: Response) => {
        logsRequest(route, req);

        let ownihrz_table = db.table("OWNIHRZ");

        const botId = req.params["bot_id"];
        const adminKey = req.params["admin_key"];

        if (!config?.api.apiToken) {
            console.log("Error: Failed to load config");
            return res.status(500).send("Failed to load config");
        }

        if (!validateAdminKey(adminKey)) {
            console.log("[Delete] Erreur admin_key n'est pas valide!");
            return res.status(403).send("Invalid admin_key!");
        };

        let ownerid1 = getOwnerByCode(await ownihrz_table.get("CLUSTER"), botId);
        let bot = await ownihrz_table.get(`CLUSTER.${ownerid1}.${botId}`);

        if (!fs.existsSync(path.join(process.cwd(), 'ownihrz', botId)) && !bot) {
            console.log("[Delete] Erreur bot_id n'existe pas!");
            return res.status(403).send("Invalid bot_id!");
        };

        [
            {
                line: `pm2 stop ${botId} -f`,
                cwd: process.cwd()
            },
            {
                line: `pm2 delete ${botId}`,
                cwd: process.cwd()
            },
            {
                line: `rm -r --interactive=never ${botId}`,
                cwd: path.join(process.cwd(), 'ownihrz')
            },
        ].forEach((index) => {
            try {
                execSync(index.line, { stdio: [0, 1, 2], cwd: index.cwd });
            } catch (e: any) {
                console.log(e.toString().split('\n')[0]);
            }
        });

        await ownihrz_table.delete(`CLUSTER.${ownerid1}.${botId}`);

        return res.sendStatus(200);
    },
};